ppl/src/main/java/org/opensearch/sql/ppl/utils/ArgumentFactory.java

```diff
@@ -148,6 +152,26 @@
             .singletonList(new Argument("k", getArgumentValue(ctx.k)));
   }
 
+  /**
+   * Get list of {@link Argument}.
+   *
+   * @param ctx ADCommandContext instance
+   * @return the list of arguments fetched from the kmeans command
+   */
+  public static Map<String, Literal> getArgumentMap(OpenSearchPPLParser.AdCommandContext ctx) {
+    return new HashMap<String, Literal>() {{
+      put("shingle_size", (ctx.shingle_size != null)
+              ? getArgumentValue(ctx.shingle_size)
+              : new Literal(8, DataType.INTEGER));
```

Comment:
We already have these default values in MLCommons. How about we just set these values as null in PPL, that means we just let MLCommons to decide what's the default value. It may conflict in future if we have both PPL and MLCommons set default values.